proxy/proxy.go

```diff
@@ -3,82 +3,90 @@ package proxy
 import (
 	"fmt"
 	"net/url"
-	"time"
+	"os"
+	"os/signal"
+	"context"
+	"syscall"
 
-	"github.com/9seconds/httransform"
+	"github.com/9seconds/httransform/v2"
+	"github.com/9seconds/httransform/v2/auth"
+	"github.com/9seconds/httransform/v2/layers"
+	"github.com/9seconds/httransform/v2/dialers"
+	"github.com/9seconds/httransform/v2/executor"
 
 	"github.com/scrapinghub/crawlera-headless-proxy/config"
-	"github.com/scrapinghub/crawlera-headless-proxy/layers"
+	customs "github.com/scrapinghub/crawlera-headless-proxy/layers"
 	"github.com/scrapinghub/crawlera-headless-proxy/stats"
 )
 
 func NewProxy(conf *config.Config, statsContainer *stats.Stats) (*httransform.Server, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		for range signals {
+			cancel()
+		}
+	}()
+
 	crawleraURL, err := url.Parse(conf.CrawleraURL())
 	if err != nil {
 		return nil, fmt.Errorf("incorrect crawlera url: %w", err)
 	}
 
-	executor, err := httransform.MakeProxyChainExecutor(crawleraURL)
-	if err != nil {
-		return nil, fmt.Errorf("cannot make proxy chain executor: %w", err)
-	}
-
-	crawleraExecutor := func(state *httransform.LayerState) {
-		startTime := time.Now()
-
-		executor(state)
-		statsContainer.NewCrawleraTime(time.Since(startTime))
-		statsContainer.NewCrawleraRequest()
-	}
+	dialer, err := dialers.DialerFromURL(dialers.Opts{}, conf.CrawleraURL())
+	exe := executor.MakeDefaultExecutor(dialer)
+	pass, _ := crawleraURL.User.Password()
```

Comment:
you are getting user and password from url, but I see currently program allows you to pass apikey and user as options, without passing them to url, can this be cause of my 407 error I get when testing?

Reply:
Agreed, the way it passes user:pass is puzzling. Instead of `options > CrawleraURL > crawleraURL.User.Password() > pass` it has to become `options > pass`. But that's purely stylistic, not sure it will change the content of `pass`. 
May be change of `DialerFromURL(dialers.Opts{}, conf.CrawleraURL())`  into  `DialerFromURL(dialers.Opts{}, CrawleraURL)` will help?

Reply:
This `DialerFromURL` was added for a common case. You kinda do not need it once you have apikey already parsed and put somewhere.